src/signals/incident/containers/IncidentContainer/saga.js

```diff
@@ -42,16 +48,39 @@ export function* createIncident({ incident, wizard }) {
         'Content-Type': 'application/json'
       }
     });
+
+    if (incident.image) {
+      yield put(uploadRequest(incident.image_file, result.id));
+    }
     yield put(createIncidentSuccess(result));
   } catch (err) {
     yield put(createIncidentError(err));
   }
 }
 
+export function* uploadFile(file, id) {
+  const requestURL = `${CONFIGURATION.API_ROOT}signals/signal/image/`;
+
+  const channel = yield call(createFileUploadChannel, requestURL, file, id);
+  const { progress = 0, err, success } = yield take(channel);
+  if (err) {
+    yield put(uploadFailure(file, err));
+    return;
+  }
+  if (success) {
+    yield put(uploadSuccess(file));
+    return;
+  }
+  yield put(uploadProgress(file, progress));
+}
 // Individual exports for testing
 export default function* watchIncidentContainerSaga() {
-  yield [
-    yield takeLatest(GET_CLASSIFICATION, getClassification),
-    yield takeLatest(CREATE_INCIDENT, createIncident)
-  ];
+  yield all([
+    takeLatest(GET_CLASSIFICATION, getClassification),
+    takeLatest(CREATE_INCIDENT, createIncident),
+    takeEvery(UPLOAD_REQUEST, function* upload(action) {
+      const file = action.payload;
+      yield call(uploadFile, file, action.meta.id);
+    })
```

Comment:
move  this to another function for readability

Reply:
doe ik